server/src/main/java/org/apache/cassandra/sidecar/config/yaml/CacheConfigurationImpl.java

```diff
@@ -24,20 +24,24 @@
 import org.slf4j.LoggerFactory;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.cassandra.sidecar.common.DataObjectBuilder;
 import org.apache.cassandra.sidecar.common.server.utils.MillisecondBoundConfiguration;
 import org.apache.cassandra.sidecar.config.CacheConfiguration;
-import org.jetbrains.annotations.VisibleForTesting;
 
 /**
  * Configuration class that encapsulates parameters needed for Caches
  */
 public class CacheConfigurationImpl implements CacheConfiguration
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(CacheConfigurationImpl.class);
+    private static final long DEFAULT_MAXIMUM_SIZE = 100;
+    private static final boolean DEFAULT_ENABLED = true;
+    private static final int DEFAULT_WARMUP_RETRIES = 5;
+    private static final MillisecondBoundConfiguration DEFAULT_WARMUP_RETRY_INTERVAL = MillisecondBoundConfiguration.parse("1s");
 
     protected MillisecondBoundConfiguration expireAfterAccess;
```

Comment:
Since we have a builder, shall we remove the set methods for other parameters?

Reply:
The setters are used when we have the legacy names in the yaml , for example `expire_after_access_millis` is the legacy property. New properties should not have a setter because they should not have a legacy yaml property